source/features/pr-branch-auto-delete.tsx

```diff
@@ -2,22 +2,25 @@ import select from 'select-dom';
 import delegate from 'delegate-it';
 import * as pageDetect from 'github-url-detection';
 
-import features from '.';
 import observeElement from '../helpers/simplified-element-observer';
+import features, {setupDeinit} from '.';
 
-function init(): void {
+function init(): Deinit {
 	const subscription = delegate(document, '.js-merge-commit-button', 'click', () => {
 		subscription.destroy();
-
-		observeElement('.discussion-timeline-actions', (_, observer) => {
+		setupDeinit(observeElement('.discussion-timeline-actions', (_, observer) => {
```

Comment:
Why `setupDeinit` being used outside the manager? This doesn't look right. This looks like a prime target for `signal` instead. Alternatively it should just be turned into a regular `MutationObserver` in the top level, which can then be regularly returned, unchanged, in the `init` return as `[observer, subscription]`

Reply:
> This looks like a prime target for `signal` instead

Sadly this is just a wrapper around a `MutationObserver` which doesn't support `AbortSignal`, so that would mean setting up a event listener on the signal.

> Alternatively it should just be turned into a regular `MutationObserver` in the top level

+1